editors/code/src/scopes_mapper.ts

```diff
@@ -0,0 +1,79 @@
+import * as vscode from 'vscode';
+import { TextMateRuleSettings } from './scopes';
+
+let mappings = new Map<string, string[]>();
+
+const defaultMapping = new Map<string, string[]>([
+    [
+        'comment',
+        [
+            'comment',
+            'comment.block',
+            'comment.line',
+            'comment.block.documentation'
+        ]
+    ],
+    ['string', ['string']],
+    ['keyword', ['keyword']],
+    ['keyword.control', ['keyword.control', 'keyword', 'keyword.other']],
+    [
+        'keyword.unsafe',
+        ['storage.modifier', 'keyword.other', 'keyword.control', 'keyword']
+    ],
+    ['function', ['entity.name.function']],
+    ['parameter', ['variable.parameter']],
+    ['constant', ['constant', 'variable']],
+    ['type', ['entity.name.type']],
+    ['builtin', ['variable.language', 'support.type', 'support.type']],
+    ['text', ['string', 'string.quoted', 'string.regexp']],
+    ['attribute', ['keyword']],
+    ['literal', ['string', 'string.quoted', 'string.regexp']],
+    ['macro', ['support.other']],
+    ['variable', ['variable']],
+    ['variable.mut', ['variable', 'storage.modifier']],
+    [
+        'field',
+        [
+            'variable.object.property',
+            'meta.field.declaration',
+            'meta.definition.property',
+            'variable.other'
+        ]
+    ],
+    ['module', ['entity.name.section', 'entity.other']]
+]);
+
+// Temporary exported for debugging for now.
```

Comment:
Is this still needed?

Reply:
@lnicola Well depends, do we want to log any issues or not?

Reply:
I'm not sure. Maybe we could log only once for each tag to make it less verbose, but it still seems annoying. I guess we can try to:

- support the built-in themes (detect them by name and have some overrides)
- document a list of extra TM scopes we're using (if any, for things like macros)
- (not sure it's a good idea: ) use the standard scopes (to have some highlighting, at least), then add out semantically-correct ones

Do you think this a good plan?

Reply:
@lnicola Sorry I missed this. 

I think the approach I took, is log what tags didn't work and just let the user know that they can extend it .

```jsonc
    "rust-analyzer.scopeMappings": {
        "keyword": [
            "keyword",
            "keyword.control",
            "storage.modifier.visibility.rust"
        ]
    },
```